src/bootstrap.php

```diff
@@ -1711,62 +1475,47 @@ function (Request $request, int $number) use ($app) {
         $content['items'][] = $package;
     }
 
-    $headers = ['Content-Type' => sprintf('%s; version=%s', $type, $version)];
+    $headers = ['Content-Type' => $type->getNormalizedValue()];
 
     return new Response(
         json_encode($content, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES),
         Response::HTTP_OK,
         $headers
     );
-});
+})->before($app['negotiate.accept'](
+    'application/vnd.elife.press-package-list+json; version=1'
+));
 
 $app->get('/press-packages/{id}',
-    function (Request $request, string $id) use ($app) {
+    function (Accept $type, string $id) use ($app) {
         if (false === isset($app['press-packages'][$id])) {
             throw new NotFoundHttpException('Not found');
         }
 
         $packages = $app['press-packages'][$id];
 
-        $accepts = [
-            'application/vnd.elife.press-package+json; version=2',
-            'application/vnd.elife.press-package+json; version=1',
-        ];
-
-        /** @var Accept $type */
-        $type = $app['negotiator']->getBest($request->headers->get('Accept'), $accepts);
-
-        $version = (int) $type->getParameter('version');
-        $type = $type->getType();
-        $headers = ['Content-Type' => sprintf('%s; version=%s', $type, $version)];
+        $headers = ['Content-Type' => $type->getNormalizedValue()];
 
-        if ($version < 2 && empty($packages['relatedContent'])) {
-            throw new UnsupportedVersion(sprintf('This press package requires version %d.', 2));
+        if ($type->getParameter('version') < 2 && empty($packages['relatedContent'])) {
+            throw new NotAcceptableHttpException(sprintf('This press package requires version %d.', 2));
         }
 
-        $latest = new Accept($accepts[0]);
-        if ($version < $latest->getParameter('version')) {
-            $headers['Warning'] = sprintf('299 elifesciences.org "Deprecation: Support for version %d will be removed"', $version);
+        if ($type->getParameter('version') < 2) {
+            $headers['Warning'] = sprintf('299 elifesciences.org "Deprecation: Support for version %d will be removed"', $type->getParameter('version'));
```

Comment:
Can this be automated somehow?

Reply:
`this` being the `Warning` headers generation, possible options:
- `response-transformer` plugin in the `api-gateway`, but it doesn't support conditions so can only add headers to all responses of an API
- nginx magic could probably map a `Content-Type` header value to a `Warning` header if it's configured with the list of deprecated values by introspecting the api-raml. These maps have limited size however so it's not very scalable nor easy to understand.
- PHP code that reads `vendor/elife/api` or some build-time transformation of that to add the headers basing on `Content-Type` when post-processing responses.

The first two could potentially be applied to other projects as well, but seem intricate for maintenance.

Reply:
Wasn't quite thing about that level, more if the negotiation here hasn't produced the latest version add the header. (We don't have any validation for implementations against the RAML, and the Warning header is probably the least of our concerns for that.)

Reply:
Could just be a middleware (at the app rather than route level as a couple of endpoints produce the same type) that just added it if the `Content-Type` is in a set list.

Reply:
:+1: for a middleware with a white list of deprecated `Content-Type` values, everything more is too much of a concern for the `Warning` header